OpenRA.Mods.Common/Traits/Render/WithDeathAnimation.cs

```diff
@@ -52,7 +76,10 @@ public void Killed(Actor self, AttackInfo e)
 
 			var sequence = Info.DeathSequence;
 			if (Info.UseDeathTypeSuffix)
-				sequence += e.Warhead.DeathType;
+			{
+				var damageType = e.Warhead.DamageTypes.Intersect(Info.DeathTypes.Keys).First();
```

Comment:
This will throw a rather incomprehensible exception if the actor is killed with an unhandled death type.  Could we make this use FirstOrDefault and then either return or (preferrably IMO) throw a descriptive exception that mentions the weapon name and the lists of damage types on the weapon and death anim dictionary?